client/components/common/BuildList.tsx

```diff
@@ -96,6 +117,51 @@ const BuildList: React.FC<Props> = ({
     DEBOUNCE_INTERVAL,
   );
 
+  // Takes params from the router and filters them out so that the inherent username value is not in the address bar
+  // twice, also when values are undefined or empty strings. Returns a new object to use in formatQueryString.
+  const formatParams = (obj: ProfileQueryParams) =>
+    Object.entries(obj)
+      .filter(([, val]) => val !== undefined && val !== '' && val !== username)
+      .reduce((result, [key, val]) => {
+        const newResult = {
+          ...result,
+          [key]: val,
+        };
+        return newResult;
+      }, {} as ProfileQueryParams);
+
+  const formatQueryString = () => {
+    const params = formatParams({
+      ...router.query,
+      search,
+      class: classData?.classes.find((item) => item.id === dofusClassId)
+        ?.enName,
+      tags: tagIds
+        .map(
+          (tagId) =>
+            tagsData?.customSetTags.find((tag) => tagId === tag.id)?.enName,
+        )
+        .join(','),
+    });
+
+    return !Object.keys(params).length
+      ? ''
+      : `?${Object.entries(params)
+          .map((item) => `${item[0]}=${item[1]}`)
+          .join('&')}`;
+  };
+
+  React.useEffect(() => {
+    if (!isProfile) {
+      return;
+    }
+    if (!didMount.current) {
+      didMount.current = true;
+    } else {
+      router.replace(`${username}${formatQueryString()}`);
```

Comment:
This seems kind of hacky. If we move the state to the query params themselves, then we shouldn't need an extra useEffect.

Reply:
This is completely possible, yeah. But then we get the encoding -for lack of a better word- issues where the URL is separated by C%2 instead of commas, which is probably not a huge thing, but does look kind of ugly, haha.

I do get it is kind of hacky since that ref is there to just ignore the first render of the component. Would you want to discuss how to go with this? Would you prefer the other encoding?